src/swift/Sources/Models/TokenFile.swift

```diff
@@ -229,465 +234,430 @@ class TokenFile: Codable {
-        guard publicModifiers.contains(decl.accessLevelModifier ?? .internal) else {
-            return
-        }
-        let value = (decl.accessLevelModifier ?? .internal).textDescription
-        newLine()
-        if !decl.attributes.isEmpty {
-            handle(attributes: decl.attributes)
-        }
-        keyword(value: value)
-        whitespace()
         keyword(value: "extension")
         whitespace()
         let defId = decl.type.textDescription
         lineIdMarker(definitionId: defId)
         type(name: decl.type.textDescription, definitionId: defId)
         whitespace()
-        if let inheritance = decl.typeInheritanceClause {
-            handle(clause: inheritance)
-        }
-        if let genericWhere = decl.genericWhereClause {
-            handle(clause: genericWhere)
-        }
+        handle(clause: decl.typeInheritanceClause)
+        handle(clause: decl.genericWhereClause)
         punctuation("{")
+        newLine()
         indent {
             decl.members.forEach { member in
-                switch member {
-                case .declaration(let decl):
-                    process(decl)
-                default:
-                    return
-                }
+                handle(member: member, overridingAccess: accessLevel)
             }
         }
-        if needsNewLine {
-            newLine()
-        }
         punctuation("}")
+        newLine()
+        return true
     }
 
-    private func process(_ decl: ConstantDeclaration) {
-        let accessLevel = decl.modifiers.accessLevel
-        var name = "NAME"
-        let isStatic = decl.modifiers.isStatic
-        var typeModel: TypeModel? = nil
-
-        decl.modifiers.verifySupported()
-
+    private func process(_ decl: ConstantDeclaration, overridingAccess: AccessLevelModifier? = nil) -> Bool {
         for item in decl.initializerList {
-            if case let identPattern as IdentifierPattern = item.pattern {
-                name = identPattern.identifier.textDescription
-                typeModel = TypeModel(from: identPattern)
+            if let typeModel = item.typeModel {
+                let accessLevel = decl.modifiers.accessLevel ?? overridingAccess ?? .internal
+                return processMember(name: item.name, attributes: decl.attributes, modifiers: decl.modifiers, typeModel: typeModel, isConst: true, defaultValue: item.defaultValue, accessLevel: accessLevel)
             }
         }
+        SharedLogger.fail("Type information not found.")
+    }
 
-        guard publicModifiers.contains(accessLevel ?? .internal) else { return }
-        newLine()
-        keyword(value: accessLevel!.textDescription)
-        whitespace()
-        if isStatic {
-            keyword(value: "static")
-            whitespace()
+    private func process(_ decl: VariableDeclaration, overridingAccess: AccessLevelModifier? = nil) -> Bool {
+        var name: String
+        var typeModel: TypeModel
```

Comment:
Ditto comment: I don't really get what the type model is doing

Reply:
https://github.com/Azure/azure-sdk-tools/pull/1820/files#r672657358

We can hop on a call if that's not clear.